tokio-trace/tokio-trace-core/src/callsite.rs

```diff
@@ -24,6 +24,41 @@ struct Registry {
     dispatchers: Vec<dispatcher::Registrar>,
 }
 
+impl Registry {
+    fn establish_interest(&self, callsite: &'static Callsite) {
```

Comment:
Since we're adding `set_interest`, I think we should just remove the `add_interest` and `clear_interest` functions, and implement both `establish` and `reestablish` in terms of `set_interest`. In that case we no longer need separate `establish` and `reestablish` functions.

Reply:
I *think* this is fine as long as implementations of `set_interest` implementations use a `SeqCst` to synchronize w/ the mutex.

Reply:
@carllerche I believe the generated callsites _currently_ use a `Relaxed` ordering since there wasn't previously a need to synchronize here; we should make sure to change this as part of this branch.